msnoise/s01scan_archive.py

```diff
@@ -50,210 +52,526 @@
+    true) and updates data availaibility information in the database.
+
+    :param init: scan all files in the archive, disregarding their modification
+        time (if not None, crondays is ignored).
+    :param threads: the maximum number of threads/processes to use while
+        scanning the files.
+    :para crondays: scan only files modified in crondays number of days in the
+        past (according to the file's modification time); if None, read its
+        value from the configuration.
+    :para forced_path: scan files from this path instead of the configured
+        archive path.
+    :para forced_path_recursive: whether to also scan files from subdirectories
+        of forced_path.
+    """
+    scanning_starttime = time.time()
     logger.info('*** Starting: Scan Archive ***')
-    db = connect()
+    db = api.connect()
 
-    mtime = get_config(db, 'crondays')
+    if api.get_tech() == 1 and threads != 1:
+        logger.warning('You can not work on %i threads because SQLite only'
+                       ' supports 1 connection at a time. Continuing with '
+                       ' 1 process only.' % threads)
+        threads = 1
+    logger.info('Will work on %i thread(s)' % threads)
 
     if init:
-        logger.info("Initializing (should be run only once)")
-        mtime = "-20000"
-    else:
-        mtime = "%s" % mtime
-
-    nthreads = 1
-    if threads:
-        nthreads = threads
-    if get_tech() == 1 and nthreads > 1:
-        logger.info("You can not work on %i threads because SQLite only"
-                     " supports 1 connection at a time" % nthreads)
-        nthreads = 1
-
-        logger.info("Will work on %i thread(s)" % nthreads)
-
-    startdate = get_config(db, 'startdate')
-    startdate = datetime.datetime.strptime(startdate, '%Y-%m-%d').date()
-    enddate = get_config(db, 'enddate')
-    enddate = datetime.datetime.strptime(enddate, '%Y-%m-%d').date()
-    logger.info("Will search for files between %s and %s"%(startdate, enddate))
-    goal_sampling_rate = float(get_config(db, "cc_sampling_rate"))
-
-    data_folder = get_config(db, 'data_folder')
-    data_struc = get_config(db, 'data_structure')
-    network = get_config(db, 'network')
-
-
-    channels = [c for c in get_config(db, 'channels').split(',')]
-    logger.debug("Will search for channels: %s" % channels)
-    folders_to_glob = []
-    if data_struc in data_structure:
-        rawpath = data_structure[data_struc]
-    elif data_struc.count('/') != 0:
-        rawpath = data_struc
+        logger.info('Initializing: updating availability using the whole'
+                    ' archive (should be run only once)')
     else:
-        logger.info("Can't parse the archive for format %s !" % data_struc)
-        logger.info("trying to import local parser (should return a station"
-                     " list)")
+        if crondays is None:
+            crondays = int(api.get_config(db, 'crondays'))
```

Comment:
With the new default.default structure, you can set the type of the variable (see above)

Reply:
You're absolutely right, that would have been better!